share/spack/templates/modules/modulefile.tcl

```diff
@@ -26,9 +26,15 @@ proc ModulesHelp { } {
 {% endblock %}
 
 {% block autoloads %}
+if {![info exists ::env(LMOD_VERSION_MAJOR)]} {
```

Comment:
On a second look, wouldn't this be creating an empty if/else statement if no autoload modules are present?

Reply:
Good catch! I need to check if there is something to autoload prior adding this if/else statement.

Reply:
I have reworked template to check if there are elements to autoload with `{% if autoload|length > 0 %}` condition.

I have also added assertion to check if `if {![info exists ::env(LMOD_VERSION_MAJOR)]} {` condition is there or not.